src/lib.rs

```diff
@@ -658,6 +674,193 @@ impl AsRawFd for LineHandle {
     }
 }
 
+/// A collection of lines that can be accesses simultaneously
+///
+/// This is a collection of lines, all from the same GPIO chip that can
+/// all be accessed simultaneously
+#[derive(Debug)]
+pub struct Lines {
+    lines: Vec<Line>,
```

Comment:
In terms of practical usage, I wonder if it might be better to be lazy about actually doing the work to get line information for all the lines in a group.  The approach taken here most closely mirrors the initial approach taken for a single Line but is probably a mismatch with the common use case for actually creating a multi-line handle.

We can still have a `fn lines()` that would return an iterator yielding `Line` instances but my gut tells me that doing that work up-front is, in most cases, going to be wasted syscalls for multi-line handles.

Reply:
Yes, you're probably right. That would likely be a lot of systems calls, when likely the most common use case would be to get the single multi-line handle, get/set the bits, and move on.

I think that modelling it as a collection of `Line` instances is still a good idea, but it begs the question... Where should the laziness come into play? At the multi-line level? Or at the single line level?

There's a pretty decent chance that the user of a single `Line` doesn't need the info either. They might want to move straight to grabbing a handle for the line without checking the info.

Should we reconsider the API for a single `Line`?  Perhaps make a `LineInfo` struct containing the flags, name, & consumer that the user could get, if desired? Maybe something like:
```
pub struct LineInfo {
    flags: LineFlags,
    name: Option<String>,
    consumer: Option<String>,
}

pub struct Line {
    chip: Arc<Box<InnerChip>>,
    offset: u32,
    info: Option<LineInfo>,  // Maybe cache it here, when requested by the user
}

impl Line {
    pub fn get_info(&self) -> Result<LineInfo> {
        ...
```

Reply:
That could work; the possibility of getting any specific piece of metadata is a bit cumbersome.  I could see having a metadata() or info() calls that returns a struct with the actual information (since we get it all at once anyway).  Having that in an option or something could work but I think it would probably be necessary in practice.

Reply:
Did you mean "probably **not** be necessary"?